samples/subsys/usb/hid-mouse/src/main.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018 qianfan Zhao
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <board.h>
+#include <device.h>
+#include <gpio.h>
+
+#include <usb/usb_device.h>
+#include <usb/class/usb_hid.h>
+
+/* change this to use another GPIO port */
+#ifdef SW0_GPIO_NAME
+#define PORT	SW0_GPIO_NAME
+#else
+#error SW0_GPIO_NAME needs to be set in board.h
+#endif
+
+/* change this to use another GPIO pin */
+#ifdef SW0_GPIO_PIN
+#define PIN     SW0_GPIO_PIN
+#else
+#error SW0_GPIO_PIN needs to be set in board.h
+#endif
+
+/*
+ * Usage Page (Desktop),              ; Generic desktop controls (01h)
+ * Usage (Mouse),                     ; Mouse (02h, application collection)
+ * Collection (Application),
+ *    Usage (Pointer),                ; Pointer (01h, physical collection)
+ *    Collection (Physical),
+ *        Usage Page (Button),        ; Button (09h)
+ *        Usage Minimum (01h),
+ *        Usage Maximum (03h),
+ *        Logical Minimum (0),
+ *        Logical Maximum (1),
+ *        Report Count (3),
+ *        Report Size (1),
+ *        Input (Variable),
+ *        Report Count (1),
+ *        Report Size (5),
+ *        Input (Constant),
+ *        Usage Page (Desktop),       ; Generic desktop controls (01h)
+ *        Usage (X),                  ; X (30h, dynamic value)
+ *        Usage (Y),                  ; Y (31h, dynamic value)
+ *        Usage (Wheel),              ; Wheel (38h, dynamic value)
+ *        Logical Minimum (-127),
+ *        Logical Maximum (127),
+ *        Report Size (8),
+ *        Report Count (3),
+ *        Input (Variable, Relative),
+ *    End Collection,
+ * End Collection
+ */
+
+static const u8_t hid_report_desc[] = {
+	0x05, 0x01, 0x09, 0x02, 0xa1, 0x01, 0x09, 0x01,
+	0xa1, 0x00, 0x05, 0x09, 0x19, 0x01, 0x29, 0x03,
+	0x15, 0x00, 0x25, 0x01, 0x95, 0x03, 0x75, 0x01,
+	0x81, 0x02, 0x95, 0x01, 0x75, 0x05, 0x81, 0x01,
+	0x05, 0x01, 0x09, 0x30, 0x09, 0x31, 0x09, 0x38,
+	0x15, 0x81, 0x25, 0x7f, 0x75, 0x08, 0x95, 0x03,
+	0x81, 0x06, 0xc0, 0xc0,
+};
+
+static int get_report_cb(struct usb_setup_packet *setup, s32_t *len,
+		  u8_t **data)
```

Comment:
Please align.

Reply:
thanks, but i don't know what the align style, could you please give me some examples ?

Reply:
Align to the first character after `(`. E.g.:
```
static int get_report_cb(struct usb_setup_packet *setup, s32_t *len,
                         u8_t **data)
```

Reply:
what's the tab's width,4 or 8?

Reply:
>what's the tab's width,4 or 8?
8

Reply:
You have already aligned button_pressed() function, please align the same way.